app/src/test/java/de/westnordost/streetcomplete/quests/opening_hours/OpeningHoursTagParserTest.kt

```diff
@@ -0,0 +1,174 @@
+        assertEquals(parseAndSave("Mo 09:00-26:00"), "Mo 09:00-02:00") //26 -> 02
+        assertEquals(parseAndSave("Th 17:30-19:30 open"), "Th 17:30-19:30") // drop explicit open
+    }
+
+    @Test
+    fun `month limit has some impact on parser output`() {
+        assertNotEquals(OpeningHoursTagParser.parse("Jan Mo 09:00-20:00"), OpeningHoursTagParser.parse("Mo 09:00-20:00"))
+    }
+    @Test
+    fun `next day rules specified as over 24 hours, and sub 24 hours are treated equally`() {
+        assertEquals(OpeningHoursTagParser.parse("Su 09:00-26:00"), OpeningHoursTagParser.parse("Su 09:00-02:00"))
+    }
+
+    @Test
+    fun `next day rules specified as over 24 hours should be formatted as below 24 hours`() {
+        val returned = OpeningHoursTagParser.parse("Su 09:00-26:00")
+        assertNotEquals(returned, null)
+        assertEquals(returned!!.size, 1)
+        assertEquals(returned[0].months, OpeningMonthsRow().months)
+        assertEquals(returned[0].weekdaysList.size, 1)
+        assertEquals(returned[0].weekdaysList[0].timeRange.start, 9 * 60)
+        assertNotEquals(returned[0].weekdaysList[0].timeRange.end, 26 * 60)
+        assertEquals(returned[0].weekdaysList[0].timeRange.end, 2 * 60)
+    }
+
+    @Test
+    fun `test full structure of a simple opening hour`() {
+        val returned = OpeningHoursTagParser.parse("Mo 09:00-20:00")
+        assertNotEquals(returned, null)
+        assertEquals(returned!!.size, 1)
+        assertEquals(returned[0].months, OpeningMonthsRow().months)
+        assertEquals(returned[0].weekdaysList.size, 1)
+        assertEquals(returned[0].weekdaysList[0].timeRange.start, 9 * 60)
+        assertEquals(returned[0].weekdaysList[0].timeRange.end, 20 * 60)
+        assertEquals(returned[0].weekdaysList[0].timeRange.isOpenEnded, false)
+        val weekData = BooleanArray(8) { false }
+        weekData.set(0, true)
+        assertEquals(returned[0].weekdaysList[0].weekdays, Weekdays(weekData))
+    }
+
+    @Test
+    fun `test full structure of a multiday opening hour with a gap`() {
+        val returned = OpeningHoursTagParser.parse("Mo-Su 09:00-12:00, 13:00-14:00")
+        assertNotEquals(returned, null)
+        assertEquals(returned!!.size, 1)
+        assertEquals(returned[0].months, OpeningMonthsRow().months)
+        assertEquals(returned[0].weekdaysList.size, 2)
+        assertEquals(returned[0].weekdaysList[0].timeRange.start, 9 * 60)
+        assertEquals(returned[0].weekdaysList[0].timeRange.end, 12 * 60)
+        assertEquals(returned[0].weekdaysList[0].timeRange.isOpenEnded, false)
+        assertEquals(returned[0].weekdaysList[1].timeRange.start, 13 * 60)
+        assertEquals(returned[0].weekdaysList[1].timeRange.end, 14 * 60)
+        assertEquals(returned[0].weekdaysList[1].timeRange.isOpenEnded, false)
+        val weekData = BooleanArray(7) { true }
+        assertEquals(returned[0].weekdaysList[0].weekdays, Weekdays(weekData))
```

Comment:
Actually, the first parameter in `assertEquals` is what is expected and the second is the actual string. Not that big of a problem, just saying.